tests/unittests/simulation/test_output_values.py

```diff
@@ -43,65 +36,82 @@ def test_component_and_flow_output_object() -> None:
+    # Patch extra output evaluation to return empty
+    # ---------------------------------------------
+    with patch(
+        "gems.simulation.extra_output.evaluate_extra_outputs_for_a_component",
+        return_value={},
+    ):
+        output = OutputValues(mock_problem)
+
+    # ------------------------
+    # Build expected OutputValues
+    # ------------------------
     test_output = OutputValues()
-    assert output != test_output, f"Output is equal to empty output: {output}"
+    assert (
+        output != test_output
+    ), "Output should not equal an empty OutputValues initially"
 
+    # Ignore component and compare
     test_output.component("component_id_test").ignore = True
     assert (
         output == test_output
-    ), f"Output differs from the expected output after 'ignore': {output}"
+    ), "Output should equal test_output after ignoring component"
 
+    # Set variable values and ignore others
     test_output.component("component_id_test").ignore = False
     test_output.component("component_id_test").var("component_var_name").value = 1.0
     test_output.component("component_id_test").var(
         "component_approx_var_name"
     ).ignore = True
-
     assert (
         output == test_output
-    ), f"Output differs from the expected after 'var_name': {output}"
+    ), "Output should match after setting variable values and ignoring others"
 
+    # Test values outside tolerance
     test_output.component("component_id_test").var(
         "component_approx_var_name"
     ).ignore = False
     test_output.component("component_id_test").var(
         "component_approx_var_name"
     ).value = 1.000_000_001
-
     assert output != test_output and not output.is_close(
         test_output
-    ), f"Output is equal to expected outside tolerance: {output}"
+    ), "Output should differ outside tolerance"
 
+    # Test values inside tolerance
     test_output.component("component_id_test").var(
         "component_approx_var_name"
     ).value = 1.000_000_000_1
-
     assert output != test_output and output.is_close(
         test_output
-    ), f"Output differs from the expected inside tolerance: {output}"
+    ), "Output should match inside tolerance"
 
+    # Add wrong variable and ignore it
     test_output.component("component_id_test").var(
         "component_approx_var_name"
     ).ignore = True
     test_output.component("component_id_test").var(
         "wrong_component_var_name"
     ).value = 1.0
-
-    assert output != test_output, f"Output is equal to wrong output: {output}"
+    assert output != test_output, "Output should differ with wrong variable"
 
     test_output.component("component_id_test").var(
         "wrong_component_var_name"
     ).ignore = True
+    assert output == test_output, "Output should match after ignoring wrong variable"
 
-    assert output == test_output, f"Output differs from expected: {output}"
-
+    # Print final output
```

Comment:
Comment not needed, `print` is pretty clear...
In general, comments for obvious code should be removed, it adds useless length. Comments are useful to understand workflows or operations that may be tricky or not that obvious